addons/mail/models/mail_thread.py

```diff
@@ -1804,7 +1804,11 @@ def message_post(self, *,
         record_name = record_name or self.display_name
 
         # Find the message's author
-        author_id, email_from = self._message_compute_author(author_id, email_from, raise_exception=True)
+        guest_id = self.env['mail.guest']
+        if 'guest' in self.env.context:
+            guest_id = self.env.context['guest'].id
```

Comment:
passing a browserecord in the context is always a bit weird (and not rpc compatible), why not an id?

Reply:
It's non-rpc compatible on purpose to prevent people from being able to set the value.

It's used in code where we want to actually know if current user is a guest, and we need to be able to trust it (not for security directly as access is checked before, but for example to tell when posting a new message if it should be considered posted as a specific guest, and which one).